x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/inference/allocation/TrainedModelAllocationClusterServiceTests.java

```diff
@@ -220,6 +221,7 @@ public void testAddRemoveAllocationNodes() {
                     .add(buildNode("ml-node-without-room", true, 1000L))
                     .add(buildNode("not-ml-node", false, ByteSizeValue.ofGb(4).getBytes()))
                     .add(buildNode("ml-node-shutting-down", true, ByteSizeValue.ofGb(4).getBytes()))
+                    .add(buildOldNode("old-versioned-ml-node-with-room", true, ByteSizeValue.ofGb(4).getBytes()))
```

Comment:
Just testing that these old nodes are not allocated to even though they are ml nodes with room.